public/components/Flyout/flyouts/createDetector.js

```diff
@@ -0,0 +1,833 @@
+        },
+      },
+      {
+        field: 'state',
+        name: 'State',
+      },
+    ];
+    return (
+      <EuiFlexGroup direction="column" gutterSize="s" style={{ paddingTop: '20px' }}>
+        <EuiFlexItem>
+          <EuiTitle size="s">
+            <h3>Features</h3>
+          </EuiTitle>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiPanel grow={false}>
+            <EuiBasicTable
+              items={items}
+              columns={columns}
+              cellProps={() => {
+                return {
+                  textOnly: true,
+                };
+              }}
+            />
+          </EuiPanel>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+}
+
+export class FilterDisplay extends Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    let filter = this.props;
+    if (filter === 'all fields are included') {
+      return (
+        <EuiText>
+          <p className="enabled">-</p>
+        </EuiText>
+      );
+    } else {
+      return <EuiText>{filter}</EuiText>;
+    }
+  }
+}
+const FixedWidthRow = (props) => <EuiFormRow {...props} style={{ width: '150px' }} />;
+
+async function createAndStartDetector(context) {
+  const configs = context.adConfigs;
+  const httpClient = context.httpClient;
+  try {
+    const response = await httpClient.post('../api/alerting/detectors', {
+      configs,
+    });
+    const {
+      data: {
+        ok,
+        response: { _id },
+      },
+    } = response;
+    const detectorId = _id;
+    if (ok) {
+      try {
+        const response = await httpClient.post(`../api/alerting/detectors/${detectorId}/_start`);
+        const {
+          data: {
+            ok,
+            response: { _id },
+          },
+        } = response;
+        if (ok) {
+          context.setFlyout(null);
+          context.renderStartedDetectorFlyout(configs, _id, context.queriesForOverview);
+        }
+      } catch (err) {
+        if (typeof err === 'string') throw err;
+        throw 'There was a problem starting detector';
+      }
+    }
+  } catch (err) {
+    if (typeof err === 'string') throw err;
+    throw 'There was a problem createing detector';
+  }
+}
+
+function warningOrSubduedallOut(message, color, iconType) {
+  return (
+    <EuiCallOut color={color}>
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiIcon type={iconType} />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          {' '}
+          <EuiTextColor>{message}</EuiTextColor>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiCallOut>
+  );
+}
+
+function triggerCorrectCallOut(context) {
+  if (context.startedDetector) {
+    return informationCallOut(context, calloutType);
+  }
+  let calloutType = 'notValid';
+  if (
+    Object.keys(context.failures).length == 0 &&
+    Object.keys(context.suggestedChanges).length == 0
+  ) {
+    calloutType = 'valid';
+  }
+  for (let [key, value] of Object.entries(context.suggestedChanges)) {
```

Comment:
it seems that you're looping through all of the suggested changes, and would overwrite `calloutType` if there was multiple suggested changes (one for `maxInterval`, one for `filterQueryTooSparse`). Can you confirm if UX has any thoughts on showing multiple callouts if this is the case?